osf/utils/machines.py

```diff
@@ -123,7 +123,14 @@ def save_changes(self, ev):
         self.machineable.save()
 
     def resubmission_allowed(self, ev):
-        return self.machineable.provider.reviews_workflow == Workflows.PRE_MODERATION.value
+        workflow = self.machineable.provider.reviews_workflow
+        result = any(
+            [
+                workflow == Workflows.PRE_MODERATION.value,
+                workflow == Workflows.POST_MODERATION.value and self.machineable.machine_state == 'pending'
+            ]
+        )
+        return result
```

Comment:
I see. From the triggering work flow, submit from pending and rejected are considered as "resubmission".

```
    {
        'trigger': DefaultTriggers.SUBMIT.value,
        'source': [DefaultStates.PENDING.value, DefaultStates.REJECTED.value],
        'conditions': 'resubmission_allowed',
        'dest': DefaultStates.PENDING.value,
        'after': ['save_action', 'update_last_transitioned', 'save_changes', 'notify_resubmit'],
    },
```

Can you add a brief comment to this method?